cloudinary-core/src/main/java/com/cloudinary/Signer.java

```diff
@@ -0,0 +1,36 @@
+package com.cloudinary;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class Signer {
```

Comment:
We don't need this class, as explained above. It tries to be both the data and the logic. There should just be an enum for the data. That enum can have a string value attached (in a format that `MessageDigest` recognizes).
An enum can also easily load from string - no need to write your own `getByName()`

And this class becomes a two-line method in utils (and it gets the signature algorithm enum as a parameter).